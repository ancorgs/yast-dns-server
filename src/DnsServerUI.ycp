/**
 * File:	modules/DnsServer.ycp
 * Package:	Configuration of dns-server
 * Summary:	Data for configuration of dns-server, input and output functions.
 * Authors:	Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 * Representation of the configuration of dns-server.
 * Input and output routines.
 */

{

    module "DnsServerUI";
    textdomain "dns-server";

    import "DnsServer";
    import "DnsTsigKeys";
    import "DnsZones";
    import "Mode";
    import "Popup";
    import "Progress";
    import "Report";
    import "Require";
    import "Runlevel";
    import "Summary";
    import "SuSEFirewall";

    define map<string,any> current_zone = $[];
    define list<map<string,any> > current_section = [];
    define list<string> current_zone_ns = [];
    define list<string> current_zone_mx = [];
    define list<string> current_zone_masters = [];
    define list<map<string,string> > current_zone_upd_ops = [];

    define list<map<string,string> > current_tsig_keys = [];
    define list<string> new_tsig_keys = [];
    define list<string> deleted_tsig_keys = [];

    include "dns-server/misc.ycp";
    include "dns-server/helps.ycp";
    include "dns-server/options.ycp";

    /**
      * Map of popups for CWM
      */
    global map popups = InitPopups ();

    include "dns-server/widgets.ycp";

    /**
      * Map of widgets for CWM
      */
    global map widgets = InitWidgets ();

    include "dns-server/dialogs.ycp";
    include "dns-server/wizards.ycp";

/* EOF */
}
