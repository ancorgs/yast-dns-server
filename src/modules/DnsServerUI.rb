# encoding: utf-8

# File:	modules/DnsServer.ycp
# Package:	Configuration of dns-server
# Summary:	Data for configuration of dns-server, input and output functions.
# Authors:	Jiri Srain <jsrain@suse.cz>
#
# $Id$
#
# Representation of the configuration of dns-server.
# Input and output routines.
require "yast"

module Yast
  class DnsServerUIClass < Module
    def main
      Yast.import "UI"
      textdomain "dns-server"

      Yast.import "DnsServer"
      Yast.import "DnsTsigKeys"
      Yast.import "DnsZones"
      Yast.import "Mode"
      Yast.import "Popup"
      Yast.import "Progress"
      Yast.import "Report"
      Yast.import "Summary"
      Yast.import "SuSEFirewall"

      Yast.import "DnsFakeTabs" # FIXME remove when possbile

      @current_zone = {}
      @current_section = []
      @current_zone_ns = []
      @current_zone_mx = []
      @current_zone_masters = []
      @current_zone_upd_ops = []

      @current_tsig_keys = []
      @new_tsig_keys = []
      @deleted_tsig_keys = []

      @forwarders = []
      #    string forward = "";

      @options = []
      @current_option_index = 0

      @zones = []

      @acl = []

      @logging = []

      @was_editing_zone = false

      Yast.include self, "dns-server/misc.rb"
      Yast.include self, "dns-server/helps.rb"
      Yast.include self, "dns-server/options.rb"

      # Map of popups for CWM
      @popups = InitPopups()

      Yast.include self, "dns-server/dialogs.rb"
      Yast.include self, "dns-server/dialog-main.rb"
      Yast.include self, "dns-server/dialog-masterzone.rb"
      Yast.include self, "dns-server/dialog-installwizard.rb"
      Yast.include self, "dns-server/wizards.rb" 

      # EOF
    end

    publish :function => :SetDNSSErverIcon, :type => "void ()"
    publish :function => :getZoneType, :type => "symbol (string)"
    publish :variable => :HELPS, :type => "map"
    publish :function => :globalPopupInit, :type => "void (any, string)"
    publish :function => :globalPopupStore, :type => "void (any, string)"
    publish :function => :globalTableEntrySummary, :type => "string (any, string)"
    publish :function => :masterPopupInit, :type => "void (any, string)"
    publish :function => :masterPopupStore, :type => "void (any, string)"
    publish :function => :masterTableEntrySummary, :type => "string (any, string)"
    publish :function => :masterTableLabelFunc, :type => "string (any, string)"
    publish :function => :ASummary, :type => "string (any, string)"
    publish :function => :CNAMESummary, :type => "string (any, string)"
    publish :function => :PTRSummary, :type => "string (any, string)"
    publish :function => :NSSummary, :type => "string (any, string)"
    publish :function => :MXSummary, :type => "string (any, string)"
    publish :function => :MXInit, :type => "void (any, string)"
    publish :function => :MXStore, :type => "void (any, string)"
    publish :function => :InitPopups, :type => "map ()"
    publish :variable => :popups, :type => "map"
    publish :function => :ReallyExit, :type => "boolean ()"
    publish :function => :ReallyAbort, :type => "boolean ()"
    publish :function => :ReallyAbortAlways, :type => "boolean ()"
    publish :function => :ReadDialog, :type => "symbol ()"
    publish :function => :WriteDialog, :type => "symbol ()"
    publish :function => :confirmAbort, :type => "boolean ()"
    publish :function => :confirmAbortIfChanged, :type => "boolean ()"
    publish :variable => :functions, :type => "map <symbol, any>"
    publish :function => :TabSequence, :type => "symbol ()"
    publish :function => :InstallWizardSequence, :type => "symbol ()"
    publish :function => :DnsSequence, :type => "symbol ()"
    publish :function => :DnsAutoSequence, :type => "symbol ()"
  end

  DnsServerUI = DnsServerUIClass.new
  DnsServerUI.main
end
