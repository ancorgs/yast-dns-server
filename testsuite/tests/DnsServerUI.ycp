{
    // testedfiles: DnsServerUI

    // While running tests, variable Y2ALLGLOBAL makes all functions global
    // thus we can test all internal functions

    include "testsuite.ycp";

    import "DnsServerUI";

    // DnsServerUI::ChangeIPToLocalEquivalent
    DUMP ("==========================================================");

    list <string> ips = [
	"192.168.5.1",
	"238.11.26.25",
	"fe80::21c:c0ff:fe18:f01c",
	"trash"
    ];

    foreach (string ip, ips, {
	TEST (``(DnsServerUI::ChangeIPToLocalEquivalent (ip)), [], nil);
    });

    DUMP ("==========================================================");

    map READ = $[];
    map WRITE = $[];
    map EXEC = $[
	"target" : $[
	    "bash_output" : $["exit" : 0, "stdout" : "127.0.0.1\n127.0.0.2\n::1\n192.168.5.1\n238.11.26.25\nfe80::21c:c0ff:fe18:f01c", "stderr" : ""],
	],
    ];

    // DnsServerUI::CurrentlyUsedIPs
    TEST (``(DnsServerUI::CurrentlyUsedIPs (true)), [READ, WRITE, EXEC], nil);
    TEST (``(DnsServerUI::CurrentlyUsedIPs (false)), [READ, WRITE, EXEC], nil);

    DUMP ("==========================================================");
}
