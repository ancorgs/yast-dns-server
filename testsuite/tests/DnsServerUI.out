Dump	==========================================================
Return	127.0.0.1
Return	127.0.0.1
Return	::1
Return	nil
Dump	==========================================================
Return	["127.0.0.1", "127.0.0.2", "::1", "192.168.5.1", "238.11.26.25", "fe80::21c:c0ff:fe18:f01c"]
Return	["192.168.5.1", "238.11.26.25", "fe80::21c:c0ff:fe18:f01c"]
Dump	==========================================================
